File: JavaImpl/src/main/java/org/example/BinarySearch.java
package org.example;

public class BinarySearch {

    public static int binarySearch(int[] arr, int target) {
        int left = 0, right = arr.length - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;  // نحسب مكان النص
            if (arr[mid] == target)  // العنصر موجود في النص 🎯
                return mid;
            if (arr[mid] < target)  // العنصر أكبر، يبقى في النص التاني
                left = mid + 1;
            else  // العنصر أصغر، يبقى في النص الأول
                right = mid - 1;
        }
        return -1;  // العنصر مش موجود ❌
    }

    public static void main(String[] args) {
        int[] arr = {1, 3, 5, 7, 9, 11, 13};
        System.out.println(binarySearch(arr, 7));  // النتيجة: 3 (العنصر موجود في الفهرس 3)
    }

}

File: KotlinImpl/src/main/kotlin/BinarySearch/binarySearch.kt
package org.example.BinarySearch

fun binarySearch(arr: IntArray, target: Int): Int {
    var left = 0
    var right = arr.size - 1
    while (left <= right) {
        val mid = left + (right - left) / 2  // نحسب مكان النص
        when {
            arr[mid] == target -> return mid  // العنصر موجود في النص 🎯
            arr[mid] < target -> left = mid + 1  // العنصر أكبر، يبقى في النص التاني
            else -> right = mid - 1  // العنصر أصغر، يبقى في النص الأول
        }
    }
    return -1  // العنصر مش موجود ❌
}

fun main() {
    val arr = intArrayOf(1, 3, 5, 7, 9, 11, 13)
    println(binarySearch(arr, 7))  // النتيجة: 3 (العنصر موجود في الفهرس 3)
}
